extensions/amp-story/0.1/animation-presets.js

```diff
@@ -25,6 +26,44 @@ import {
 } from './animation-presets-utils';
 import {px} from '../../../src/style';
 
+/**
+ * A list of panning animation names.
+ * @private @const {!Array<string>}
+ */
+const PANNING_ANIMATION_NAMES = [
+  'pan-up',
+  'pan-down',
+  'pan-right',
+  'pan-left',
+];
+
+/**
+ * A mapping of animation names to CSS class names.
+ * @private @const {!Object<string, string>}
+ */
+const ANIMATION_CSS_CLASS_NAMES = {
+  'pan': 'i-amphtml-story-grid-template-with-pan-animation',
```

Comment:
Does this problem only occur with pan?  i.e. Do zoom animations ever show whitespace?

Reply:
I don't think we want to apply this styling to zoom for a couple of reasons:

1. It's fair to assume that all the use cases for panning will have to do with images that outgrow the size of the viewport (from the name of the animation), but I'm not so sure for zooming. e.g. Users may want to zoom in on an object that just takes a fraction of the screen.
2. If the fill layout resizes the image to be the same size of the viewport, that wouldn't break the animation, since zoom doesn't depend on the image dimensions. It's just a simple scale.

WDYT?

Reply:
Thinking about it more, we might want to also include zoom animations. It will make it more straight forward for when users want to combine both the pan and the zoom. This way they won't have to wrap everything inside a div. I'll do a quick prototype and see if it works.

Reply:
SGTM; I think especially if it's a child of a `template="fill"` layer, we might want this.

Reply:
Prototyped it with the zooming and it works well. I think we should also include it since it eliminates the need to wrap images inside a div when using multiple images for the animation.